From d32a86f99eb1748ba91e498e7bdfaf6657d0f6d1 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Mon, 11 Dec 2023 12:09:41 +0100
Subject: [PATCH 0718/1141] pinctrl: stm32: move PM to noirq

PCIe driver needs pinctrl and gpio available during resume_noirq()
to properly operate.

Move pinctrl driver from late to noirq.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Reported-by: Christian Bruel <christian.bruel@foss.st.com>
Change-Id: Ieab0b4c0d02449c1d794f641fa84e94e6424d859
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/345427
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Tested-by: Christian BRUEL <christian.bruel@foss.st.com>
Reviewed-by: Christian BRUEL <christian.bruel@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 drivers/pinctrl/stm32/pinctrl-stm32mp257.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pinctrl/stm32/pinctrl-stm32mp257.c b/drivers/pinctrl/stm32/pinctrl-stm32mp257.c
index 6709bddd9718..588fb5249f9c 100644
--- a/drivers/pinctrl/stm32/pinctrl-stm32mp257.c
+++ b/drivers/pinctrl/stm32/pinctrl-stm32mp257.c
@@ -2570,7 +2570,7 @@ static const struct of_device_id stm32mp257_pctrl_match[] = {
 MODULE_DEVICE_TABLE(of, stm32mp257_pctrl_match);
 
 static const struct dev_pm_ops stm32_pinctrl_dev_pm_ops = {
-	 SET_LATE_SYSTEM_SLEEP_PM_OPS(stm32_pinctrl_suspend, stm32_pinctrl_resume)
+	 NOIRQ_SYSTEM_SLEEP_PM_OPS(stm32_pinctrl_suspend, stm32_pinctrl_resume)
 };
 
 static struct platform_driver stm32mp257_pinctrl_driver = {
-- 
2.39.2

